control-plane/api/mesh/v2beta1/gateway_class_types.go

```diff
@@ -50,97 +37,23 @@ type GatewayClassList struct {
 	Items           []*GatewayClass `json:"items"`
 }
 
-func (in *GatewayClass) ResourceID(namespace, partition string) *pbresource.ID {
```

Comment:
These funcs all relate to write/reading the type from Consul; however, we don't and will not be doing that. Removing the interface implementation helps prevent someone from accidentally trying to write this type through to Consul